libimage/runtime.go

```diff
@@ -389,15 +389,16 @@ func (r *Runtime) lookupImageInDigestsAndRepoTags(name string, options *LookupIm
 		return nil, "", err
 	}
 
-	if !shortnames.IsShortName(name) {
-		named, err := reference.ParseNormalizedNamed(name)
-		if err != nil {
-			return nil, "", err
-		}
-		digested, hasDigest := named.(reference.Digested)
-		if !hasDigest {
+	named, err := reference.ParseNormalizedNamed(name)
+	if err != nil {
+		return nil, "", err
+	}
+	digested, hasDigest := named.(reference.Digested)
+	if !hasDigest {
+		if !shortnames.IsShortName(name) {
```

Comment:
This is a bit weird location for an exit.

Isn’t this effectively
```go
if digested {
   if image, found := findByDigest(); found { return image }
}
if shortName {
  if image, found := findByLaxRepoMatching(); found { return image }
}
fail errImageUnknown
```
? That is a bit more regular structure for a set of heuristics, and the “not found” error value is not copy&pasted in two places.

Reply:
Much cleaner, thank you.